### ISSUE TITLE
RFC: whois telegraf input plugin

### ISSUE THREAD

**Issue by zabullet** [influxdata/telegraf#7893]
## Feature Request

I've been thinking about monitoring domain registrations on our fleet (specifically expiration). I was thinking of using the _whois_ command with the exec plugin, but that creates a dependency on _whois_ (which isn't installed as default on Ubuntu)

### Proposal:

Build a specific whois plugin using https://github.com/domainr/whois to query a set of domains

### Use case

Detecting expiration of domain names
Mapping domains to registrars
Tracking NS servers

**Comment by ssoroka** [comment]
Check out the [x509_cert](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/x509_cert) plugin. I think this covers at least some of your use cases.

There's also the [reverse dns](https://github.com/influxdata/telegraf/tree/master/plugins/processors/reverse_dns) lookup plugin; but I don't think that's exactly what you're looking for.

**Comment by zabullet** [comment]
Thanks @ssoroka, but neither the _X509_cert_ nor _reverse dns_ plugin cover any of my use cases.

**Comment by ssoroka** [comment]
fair enough. In the mean time, you could probably build a prototype with inputs.execd (service) or inputs.exec (script) plugins.

**Comment by zabullet** [comment]
@ssoroka I appreciate the input, but I've already stated that as an option and why I don't want to use that method. Do you have anything to add about the actual proposal?

**Comment by ssoroka** [comment]
The proposal is good. It looks like https://github.com/domainr/whois is a bit immature (open issue about it not following whois redirects), but this is probably not a huge issue. I imagine something like this running once a day would be useful.

I'd be happy to accept a plugin like this.

**Comment by UAnton** [comment]
Any news or variants on how to monitoring domain expiration?

**Comment by powersj** [comment]
next steps: look into the above library to collect metrics, like # of seconds till expire, on domains. This is going to be similar logic, but for a different domain to the x509 plugin.
